src/core/Akka.Persistence/Journal/AsyncWriteJournal.cs

```diff
@@ -217,22 +217,33 @@ protected bool ReceiveWriteJournal(object message)
         private void HandleDeleteMessagesTo(DeleteMessagesTo message)
         {
             var eventStream = Context.System.EventStream;
-            _breaker.WithCircuitBreaker(() => DeleteMessagesToAsync(message.PersistenceId, message.ToSequenceNr))
-                .ContinueWith(t => !t.IsFaulted && !t.IsCanceled
-                        ? new DeleteMessagesSuccess(message.ToSequenceNr) as object
-                        : new DeleteMessagesFailure(
-                            t.IsFaulted
-                                ? TryUnwrapException(t.Exception)
-                                : new OperationCanceledException(
-                                    "DeleteMessagesToAsync canceled, possibly due to timing out."),
-                            message.ToSequenceNr),
-                    _continuationOptions)
-                .PipeTo(message.PersistentActor)
-                .ContinueWith(t =>
+            var self = Context.Self;
+
+            async Task ProcessDelete()
```

Comment:
Replace with `async Task` local function that doesn't get `await`-ed from the outside.